spec/models/conference_spec.rb

```diff
@@ -1639,4 +1639,26 @@
       expect(free_ticket.price_cents).to eq(0)
     end
   end
+
+  describe 'after_update' do
+    let(:conference) { create(:conference) }
+    let(:first_scheduled_event) { create(:event_scheduled, program: conference.program, hour: conference.start_date + conference.start_hour.hours) }
+    let(:second_scheduled_event) { create(:event_scheduled, program: conference.program, hour: conference.start_date + conference.end_hour.hours + 1.hour) }
+
+    it 'delete event schedules that are not in hour ranges, when conference start hour is updated' do
+      first_scheduled_event
+      conference.start_hour = conference.start_hour + 1.hour
+      conference.save
+      conference.run_callbacks :update
+      expect(EventSchedule.count).to eq(0)
```

Comment:
what about using `to_change` instead of to `eq`? Then we know that the EventSchedules where really deleted.